package/cloudshell/orch/training/services/student_links.py

```diff
@@ -21,4 +23,9 @@ def _format_student_link(self, sandbox_id: str, token: str) -> str:
 
     def _create_token(self, user: str, domain: str) -> str:
         admin_token = self._sandbox_api.login()
-        return self._sandbox_api.create_token(admin_token, user, domain)
\ No newline at end of file
+        try:
+            created_token = self._sandbox_api.create_token(admin_token, user, domain)
+        except CloudShellAPIError as exc:
+            self._sandbox.logger.exception(f"Creating trainee token for {user} failed - exception occurred")
+            raise exc
```

Comment:
it should be "raise" and not "raise exc" otherwise you loos the original stack trace